custom_components/localtuya/cover.py

```diff
@@ -26,14 +34,23 @@
 DEFAULT_OPEN_CMD = "on"
 DEFAULT_CLOSE_CMD = "off"
 DEFAULT_STOP_CMD = "stop"
+DEFAULT_POSITIONING_MODE = CONF_MODE_NONE
+DEFAULT_SPAN_TIME = 25.0
 
 
 def flow_schema(dps):
     """Return schema used in config flow."""
     return {
-        vol.Optional(CONF_OPEN_CMD, default=DEFAULT_OPEN_CMD): str,
-        vol.Optional(CONF_CLOSE_CMD, default=DEFAULT_CLOSE_CMD): str,
-        vol.Optional(CONF_STOP_CMD, default=DEFAULT_STOP_CMD): str,
+        vol.Optional(CONF_OPEN_CMD, default=DEFAULT_OPEN_CMD): vol.In(["on", "open"]),
+        vol.Optional(CONF_CLOSE_CMD, default=DEFAULT_CLOSE_CMD): vol.In(
+            ["off", "close"]
```

Comment:
We can do this of course, it's good to validate properly. But do we know there are no other commands? We can of course wait for issues about it and add as needed.

Reply:
I don't know and I can't imagine, but I believe it is more safe (and comfortable for the configuration) to have these as preset instead of needing to type them.

Reply:
If we know the values already, that is better of course. Otherwise we can leave the margin of error on the user with the benefit of flexibility (by allowing custom values). But it's probably just a handful of values.

Reply:
Yep, and for the sake of speed (in case someone has to configure dozens of devices, here comes my laziness again 🤣 ) I'd be considering to have a single configuration for "on/off" vs. "open/close" couples of commands since I believe they are linked together... agree?

Reply:
Hmm, I'm not sure I understand what you mean? 😎

Reply:
Merge the config options into a thing like "CONF_OPENCLOSE_CMD" with allowed values "on_off" / "open_close", and adapt the code coherently. Just one setting instead of two (and the lazy part of me celebrates 😆 ).

Reply:
A better way I would say would be to suggest defaults (I've been thinking about it before, but not prioritized it). Since we generate the input fields/schema in `flow_schema`, we can also provide whatever defaults we want. All the datapoints and values are passed in, so we can just scan those values and see if we find "on" or "open" and pre-fill whichever we find. It's unlikely we have overlapping datapoints at the same time.

Reply:
Mmm, this time it is I to be not sure to have understood what you mean... 🙄

Reply:
The variable `dps` that is passed in to `flow_schema` is a list of strings with all supported datapoints for the device. Lets assume DP 6 is used to indicate open/on, you would find `6 (value: open)` as an example in there. We would then iterate the list (for each config option) and look for any of known values. So, for `CONF_OPEN_CMD` we would look at all datapoint, searching for `value: open` and `value: on`. Since we will find `value: open` according to example above, we can use that as a default value. It's not perfect but probably work good enough.

Reply:
Sorry but I don't get this... on/open are commands that are supposed to be sent to the dps, not read from a dps (i.e., there si no DPS 6 with "open" value). So, unless the over is already opening/closing during the configuration, you would not read the info anywhere. Am I understanding badly?

Reply:
Ah, right, my bad. This only works for "state" variables. Is the state of the cover just a bool then? Or could we use that to determine what to use?

Reply:
No, it returns "stop", "on/open" or "off/close". But when it is not operating it is "stop", of course. So in order to detect it you should go to the device and manually press the open or close buttons (assuming there is a manual control, of course). Not very comfortable...

Reply:
Nah, that's not good enough. Too much work to implement that, not worth it.